### PR TITLE
update to babel6

### PR DESCRIPTION
**Pull request by iam4x**
Need for theses

* https://github.com/babel/babel-eslint/issues/199
* https://github.com/gaearon/babel-plugin-react-transform/issues/46

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -1,20 +1,23 @@
 {
-  "stage": 0,
+  "presets": ["es2015", "react", "stage-0"],
   "env": {
     "browser": {
       "plugins": ["react-transform"],
```
Comment (justingreenberg) [inline .babelrc]: the `extra` key has been removed from config obj in babel 6. when options to plugin array in `env.browser`, wrap it [in an array and provide the options object](https://babeljs.io/docs/plugins/#options) as the second item... ie:

```javascript
"plugins": [
  ["react-transform", {
    "transforms": [{
      "transform": "react-transform-hmr",
      "imports": ["react"],
      "locals": ["module"]
    }, {
      "transform": "react-transform-catch-errors",
      "imports": [
        "react",
        "redbox-react"
      ]
    }]
  }]
]
```

File: server/index.js [hunk 2]
```diff
@@ -8,7 +8,7 @@ process.env.NODE_PATH = 'app';
 require('module').Module._initPaths();
 
 // Install `babel` hook for ES6
-require('babel/register');
+require('babel-core/register');
```
Comment (iam4x) [inline server/index.js]: We need babel-eslint to be updated first.

File: .babelrc (hunk 1 above)
Comment (iam4x) [inline .babelrc]: Yes thank's @justingreenberg, someone published a preset for this: https://github.com/danmartinez101/babel-preset-react-hmre

I'm waiting for babel-plugin-react-transform tu be updated first to make the choice :+1:

**Comment by iam4x** [comment]
I've given it a new try, but still getting shit loads of errors. There's many open issues...

**Comment by beckend** [comment]
You can use "babel-plugin-add-module-export" so you can require modules without .default property.

**Comment by MaxSvargal** [comment]
I have a solution with packages:
`   "babel": "^6.3.26",
    "babel-core": "^6.4.5",
    "babel-loader": "^6.2.1",
    "babel-polyfill": "^6.3.14",
    "babel-preset-es2015": "^6.3.13",
    "babel-preset-react": "^6.3.13",
    "babel-preset-stage-0": "^6.3.13",
    "babel-register": "^6.4.3",
    "babel-runtime": "^6.3.19" `
dev:
`   "babel-plugin-add-module-exports": "^0.1.2",
    "babel-plugin-react-transform": "^2.0.0",
    "babel-plugin-transform-decorators-legacy": "^1.3.4" `

Upgrade to latest linter and get a some little fixes ;)

**Comment by iam4x** [comment]
Finally :ship: :tada:

**Pull request merged**